### DISCUSSION TITLE
Change Default value for shape properties

### DISCUSSION THREAD

**Discussion by gitubbi** [excalidraw/excalidraw#6563] [Q&A]
Hello,

Could you please tell me if it is possible with the api or something else to change the default value of the different properties of a shape in the left menu ? 

For exampe, when we start Excalidraw, the sloppiness is set by default to "Artist", but is it possible to change it to "Architect" ?

![sloppiness](https://user-images.githubusercontent.com/132272968/236582505-997f9d0a-1db1-457e-b8ed-0d13887a1599.png)

Of course if we change it by hand, Excalidraw remenber this choice for the next shape, but I wonder if it is possible to change this default choice to have it immediatly at start.

Thank you

**Comment by gitubbi** [comment]
Looking in the json file saved by Excali, i found that in fact there is a lot of properties you can use to select the default value of the tools.

Add them to the appState of initialData when you start Excali :

```
initialData : {
      appState : {
                 currentItemStrokeColor: '#000000', 
                 currentItemStrokeStyle: 'solid',
                 currentItemStrokeWidth: 1,
                 currentItemBackgroundColor: 'transparent',
                 currentItemFillStyle: 'solid',
                 currentItemFontFamily: 1,
                 currentItemFontSize: 20,
                 currentItemTextAlign: 'left',                 
                 currentItemRoughness: 0,         //Sloppiness
                 currentItemRoundness: 'sharp',   //Edges
                 currentItemOpacity: 100,
                 currentItemEndArrowhead: 'triangle',
                 currentChartType: 'line'
     }
}
```
Sorry if it is already written somewhere in the docs.

**Reply by crapthings** [reply]
do you find how to add new button to property panel?

<img width="633" alt="image" src="https://github.com/excalidraw/excalidraw/assets/1147704/9f3d10d3-a965-4aab-b8d4-54aaeac3f406">

i want to add strokeWidth to panel

**Comment by gitubbi** [comment]
Hi crapthings.

I didn't whised to add a new button, but select the default one between the 3 states already proposed by excalidraw.

You can modify the value of the width with currentItemStrokeWidth in initialData, but it will be applied only at start and this value is not associated to one of the 3 buttons. So if you click on one of them, the value will be changed, and your custom value is definitely lost. 

In the docs i didn't see some API that allows to add custom button on this toolbar. It is possible to customize the MainMenu and the Library and some others, but not the shape propertie toolbar I think : 

https://docs.excalidraw.com/docs/@excalidraw/excalidraw/api/children-components

`We have only recently started migrating to this type of component API. Some UI components are still using render props, and some UI customization isn't supported yet (such as the toolbar or the element properties panel). Stay tuned for more updates!`

You can't add button to DOM by appending some html elements at load, because this menu is created only when you click on some shape. And anyway there is no API to add a custom click event that would apply the new stroke-width to the shape. 

Looking inside the source code, there is a function that does it, but i'm not sure it is exposed by ExcalidrawLib class (i'm not using React).

So apparently it is not possible for the moment, you can only change the default value at start as you show in your picture.

**Comment by brunotdantas** [comment]
You can use the template for that, in my example i'm changing the font to normal instead of hand-drawn

1- Create a template file
![image](https://github.com/excalidraw/excalidraw/assets/19207320/fe12d34c-003f-4af5-85ca-e68dcd04a674)

2- Open this template file and create a new text and set the style (this works for every component like new rectangles)
![image](https://github.com/excalidraw/excalidraw/assets/19207320/fb6f8b25-fefb-44c1-a579-fabe55729fe9)

3- Now test your new drawing with the default value changed

**Reply by kn107** [reply]
Hi, do you know if this is possible in the web interface https://excalidraw.com also? My issue is that the line width of the drawing tool goes back to medium whenever I reset the canvas (I want the line width to always be small).
